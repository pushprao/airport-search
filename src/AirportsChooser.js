import React, {useState} from 'react';
import './App.css';
import Button from "@material-ui/core/Button";
import Grid from "@material-ui/core/Grid";
import {makeStyles} from "@material-ui/styles";
import AirportsDropDown from "./AirportsDropDown";

const useStyles = makeStyles({
    container: {
        background: 'linear-gradient(to top, #30cfd0 0%, #330867 100%)',
        backgroundColor: '#d53369',
        backgroundImage: 'linear-gradient(to right, #cbad6d, #d53369)'
    },
    button: {
        background: 'linear-gradient(to top left, #993366 0%, #ff6666 100%)',
        border: 0,
        borderRadius: 6,
        height: 40,
        padding: '0 30px',
    }
})

function AirportsChooser() {
    // Airport selection mode is initially false
    const [selectionMode, setSelectionMode] = useState(false);

    const styles = useStyles()
    return (
        <Grid
            container
            spacing={0}
            align={"center"}
            justify={"center"}
            direction={"column"}
            className={styles.container}
        >
            {/*Airport chooser button*/}
            {!selectionMode &&
            <Button variant="contained" color="primary"
                    className={styles.button}
                    onClick={() => {
                        setSelectionMode(true)
                    }}>
                Airport Chooser
            </Button>}

            {/*Render airport selection dropdown*/}
            {selectionMode && <AirportsDropDown/>}
        </Grid>
    );
}

export default AirportsChooser;
