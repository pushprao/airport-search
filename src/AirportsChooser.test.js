import React from 'react';
import renderer from 'react-test-renderer';
import AirportsChooser from "./AirportsChooser";

it('renders airports chooser', () => {
    const tree = renderer
        .create(<AirportsChooser/>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
