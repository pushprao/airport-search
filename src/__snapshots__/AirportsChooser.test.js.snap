// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders airports chooser 1`] = `
<div
  align="center"
  className="MuiGrid-root makeStyles-container-1 MuiGrid-container MuiGrid-direction-xs-column MuiGrid-justify-xs-center"
>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-2 MuiButton-containedPrimary"
    disabled={false}
    id="airport-chooser-btn"
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      Airport Chooser
    </span>
  </button>
</div>
`;
